xbmc/Application.cpp

```diff
@@ -3085,10 +3093,15 @@ bool CApplication::PlayFile(CFileItem item, const std::string& player, bool bRes
       CVideoDatabase dbs;
       dbs.Open();
 
-      if( item.m_lStartOffset == STARTOFFSET_RESUME )
+      if( item.m_lStartOffset == STARTOFFSET_RESUME || !savedPlayerState.empty())
```

Comment:
IMO we could move the handling out of this block into it's own. And could you explain why you clear the `savedplayerstate` property?

```c++
if (item.HasProperty("savedplayerstate"))
{
  options.starttime = CUtil::ConvertMilliSecsToSecs(item.m_lStartOffset);
  options.state = item.GetProperty("savedplayerstate").asString();
}
else if (item.m_lStartOffset == STARTOFFSET_RESUME)
{
...
}
```

Reply:
clearing is needed, because from this point on we don't need it anymore. I want to avoid that it stays there for the next playback (which might get started again with this playerstate because of this not being cleared)

Reply:
> IMO we could move the handling out of this block into it's own

what do you mean with this?

Reply:
"StartPercent" is used the same and will be not cleared.

Reply:
> what do you mean with this?

there is not need to put this block inside the `if (item.m_lStartOffset == STARTOFFSET_RESUME)`, this is a separate case.

Reply:
> clearing is needed, because from this point on we don't need it anymore. I want to avoid that it stays there for the next playback (which might get started again with this playerstate because of this not being cleared)

np .. leave it if it's really necessary.

Reply:
> this is a separate case.

You showed yourself that in case of (iso) stack parts the offset gets set to START_OFFSET_RESUME if the offset is greater than zero. So in fact it is the same case - the only difference is that we don't fetch the resume point from the DB but from the saved offset and playerstate. So my preference is to keep it the same case (which, in fact it is, since we resume playback - the reason is just somewhat different).

Reply:
Ok

Reply:
> "StartPercent" is used the same and will be not cleared.

I see that now. And with item being passed by value, it probably doesn't make sense clearing it since we would operate on a copy. I'll take a look where exactly the "startpercent" property gets set